### ISSUE TITLE
ChefSpec should not load configuration files

### ISSUE THREAD

**Issue by mcquin** [sethvargo/chefspec#718]
ChefSpec runners load configuration files. This leads to unexpected behavior (see: https://github.com/sethvargo/chefspec/issues/716) and can vary across individuals' development environments. It may be preferable for ChefSpec not to load any configuration files.

**Comment by sethvargo** [comment]
It's not ChefSpec, it's Chef. Is there a way to tell Chef not to load knife.rb and stuff?

**Comment by mcquin** [comment]
No. Chef always loads configuration files. I don't think an option to not load a configuration file makes sense for Chef, but if that's the way you see forward we can try submitting a PR or RFC.

**Comment by sethvargo** [comment]
Okay, sounds good. I'll do that later 😄 

Since there's no way to disable this, I don't think there's anything I can do in ChefSpec - do you agree?

**Comment by mcquin** [comment]
I suggest we leave this issue open until the proposed change gets accepted, for tracking purposes. If Chef won't accept it then it would be good to find a workaround on this side.

**Comment by mcquin** [comment]
At the very least, some kind of "Your configuration files will be loaded" messaging would be nice.

**Comment by mcquin** [comment]
As I take another look at this, maybe there is something ChefSpec can do. It appears that the loaded config is overriding any config options I set when I initialize a runner. I would expect the runner options to take precedence over any config options.

**Comment by mivok** [comment]
For anyone else coming across this, one possible culprit is the inclusion of `chefspec/berkshelf`, which will try to load the chef config in order to provide some defaults for various berkshelf config options.

You can verify this by adding `Chef::Log.warn(caller.join("\n"))` to your chef/knife config, and running rspec. It will then dump a backtrace when the file is loaded, showing the exact code path that resulted in the config file being loaded.

If the issue is berkshelf loading the config file, then you can set `ENV['BERKSHELF_CHEF_CONFIG'] = '/dev/null'` before any `require 'chefspec/berkshelf'` line in your spec_helper.rb file, which will work around the issue by telling berkshelf your chef config file doesn't exist.
